app/helpers/locales_helper.rb

```diff
@@ -1,74 +1,80 @@
-#require 'http_accept_language'
-
 module LocalesHelper
-
-  def process_locale
-    I18n.locale = best_locale
+  def use_preferred_locale
+    I18n.locale = preferred_locale
     yield if block_given?
     save_detected_locale
   end
 
-  def selectable_locales
-    Loomio::I18n::SELECTABLE_LOCALES
+  def preferred_locale
+    strict_filter(params[:locale]) ||
+    fuzzy_filter(user_selected_locale) ||
```

Comment:
Hey, do you think we could do something like

```
strict_filter(params[:locale]) ||
fuzzy_filter([
  current_user.selected_locale,
  browser_detected_locales,
  current_user.detected_locale,
  I18n.default_locale
].flatten)
```

so that we only have to iterate through one array?